examples/rust/providers/blank-slate/src/provider.rs

```diff
@@ -0,0 +1,154 @@
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use anyhow::Context as _;
+use tokio::sync::RwLock;
+use tracing::debug;
+use wasmcloud_provider_sdk::core::HostData;
+use wasmcloud_provider_sdk::{load_host_data, run_provider, LinkConfig, Provider};
+
+use crate::config::ProviderConfig;
+
+wit_bindgen_wrpc::generate!();
+
+#[derive(Default, Clone)]
+/// Your provider struct is where you can store any state or configuration that your provider needs to keep track of.
+pub struct BlankSlateProvider {
+    #[allow(unused)]
+    config: ProviderConfig,
+    /// All components linked to this provider and their config.
+    linked_from: Arc<RwLock<HashMap<String, HashMap<String, String>>>>,
+    /// All components this provider is linked to and their config
+    linked_to: Arc<RwLock<HashMap<String, HashMap<String, String>>>>,
+}
+
+/// This `impl` block is where you can implement additional methods for your provider. We've provided two examples
+/// to run and load [`HostData`], and when you have custom logic to implement, you can add it here.
+impl BlankSlateProvider {
+    /// Execute the provider, loading [`HostData`] from the host which includes the provider's configuration and
+    /// information about the host. Once you use the passed configuration to construct a [`BlankSlateProvider`],
+    /// you can run the provider by calling `run_provider` and then serving the provider's exports on the proper
+    /// RPC topics via `wrpc::serve`.
+    ///
+    /// This step is essentially the same for every provider, and you shouldn't need to modify this function.
+    pub async fn run() -> anyhow::Result<()> {
+        let host_data = load_host_data().context("failed to load host data")?;
+        let provider = Self::from_host_data(host_data);
+        let shutdown = run_provider(provider.clone(), "blank-slate-provider")
+            .await
+            .context("failed to run provider")?;
+
+        // If your provider has no exports, simply await the shutdown to keep the provider running
+        shutdown.await;
+        Ok(())
+
+        // If your provider has exports, uncomment the following connection and serve() call instead of
+        // `shutdown.await`. This will set up RPC topics for your provider's exports and await invocations.
+        // let connection = wasmcloud_provider_sdk::get_connection();
+        // serve(
+        //     &connection.get_wrpc_client(connection.provider_key()),
+        //     provider,
+        //     shutdown,
+        // )
+        // .await
+    }
+
+    /// Build a [`BlankSlateProvider`] from [`HostData`]
+    pub fn from_host_data(host_data: &HostData) -> BlankSlateProvider {
+        let config = ProviderConfig::from(&host_data.config);
+        BlankSlateProvider {
+            config,
+            ..Default::default()
+        }
+    }
+}
+
+impl Provider for BlankSlateProvider {
+    /// When your provider is linked to a component, this method will be called with the [`LinkConfig`] that
```

Comment:
I'm biased but I'd love an example/little intro method for `init()` here

Reply:
Oh! Good idea